docs/data-analysis.md

```diff
@@ -6,6 +6,125 @@ A good way for this is using Jupyter (notebook or lab) - which provides an inter
 
+## Strategy development problem analysis
+
+Debugging a strategy (are there no buy signals, ...) can be very time-consuming.
+FreqTrade tries to help you by exposing a few helper-functions, which can be very handy.
+
+I recommend using Juptyer Notebooks for this analysis, since it offers a dynamic way to rerun certain parts.
+
+The following is a full code-snippet, which will be explained by both comments, and step by step below.
+
+```python
+# Some necessary imports
+from pathlib import Path
+
+from freqtrade.data.history import load_pair_history
+from freqtrade.resolvers import StrategyResolver
+# Define some constants
+ticker_interval = "5m"
+
+# Name of the strategy class
+strategyname = 'Awesomestrategy'
+# Location of the strategy
+strategy_location = '../xmatt/strategies'
+# Location of the data
+data_location = '../freqtrade/user_data/data/binance/'
+# Only use one pair here
+pair = "XRP_ETH"
+
+# Load data
+bt_data = load_pair_history(datadir=Path(data_location),
+                            ticker_interval = ticker_interval,
+                            pair=pair)
+print(len(bt_data))
+
+
+# Load strategy - best done in a new cell
+# Needs to be ran each time the strategy-file is changed.
+strategy = StrategyResolver({'strategy': strategyname,
+                            'user_data_dir': Path.cwd(),
+                            'strategy_path': location}).strategy
+
+# Run strategy (just like in backtesting)
+df = strategy.analyze_ticker(bt_data, {'pair': pair})
+print(f"Generated {df['buy'].sum()} buy signals")
+
+# Reindex data to be "nicer" and show data
+data = df.set_index('date', drop=True)
+data.tail()
+
+```
+
+### Explanation
+
+#### Imports and constant definition
+
+``` python
+# Some necessary imports
+from pathlib import Path
+
+from freqtrade.data.history import load_pair_history
+from freqtrade.resolvers import StrategyResolver
+# Define some constants
+ticker_interval = "5m"
+
+# Name of the strategy class
+strategyname = 'Awesomestrategy'
+# Location of the strategy
+strategy_location = 'user_data/strategies'
+# Location of the data
+data_location = 'user_data/data/binance'
+# Only use one pair here
+pair = "XRP_ETH"
+```
+
+This first section imports necessary modules, and defines some constants you'll probably need to adjust for your case.
+
+#### Load candles
+
+``` python
+# Load data
+bt_data = load_pair_history(datadir=Path(data_location),
+                            ticker_interval = ticker_interval,
+                            pair=pair)
+print(len(bt_data))
+```
+
+This second section loads the historic data and prints the amount of candles in the data.
+
+#### Run strategy and analyze results
+
+Now, it's time to load and run your strategy.
+For this, I recommend using a new cell in your notebook, since you'll want to repeat this until you're satisfied with your strategy.
+
+``` python
+# Load strategy - best done in a new cell
+# Needs to be ran each time the strategy-file is changed.
```

Comment:
"be running"?

Reply:
you've never used jupyter notebooks, did you? 😄 
you can rerun certain cells (only containing parts of the code) interactively - so i think "Needs to be ran" is correct, since it's an "order".

Reply:
> you've never used jupyter notebooks, did you? 😄
> you can rerun certain cells (only containing parts of the code) interactively - so i think "Needs to be ran" is correct, since it's an "order".

I don't even know what it is... 😁 

I use vi (the old-school one, not vim -- I'm too lazy to study extra key commands provided by vim) for text editing (pretty comfortable with it, I know I'll find it on every posix system) and grep for searching for appearances of text strings, class members, etc, in the project directory. So mypy and flake are my best friends because they make python texts more monopenisual... 😁 

Okay, let this be in this form, nevermind...

Reply:
i use vi(m) too on servers or for quick edits, vs code for major editing, atom as a quick editor.

Jupyter notebooks are a good way to quickly run python code and see the output right away - then modify that again, and look at the output again. Yes, a REPL can do the same, but not with graphics and "nice" formatted panda output :smile: